### PR TITLE
Stock spec logger

### PR DESCRIPTION
**Pull request by gvszg**

### CODE REVIEW

File: app/models/stock_spec.rb [hunk 1]
```diff
@@ -39,4 +40,12 @@ def set_on_shelf_and_send_wish_list_notification_when_amout_larger_than_zero
   def send_wish_list_notification_if_stock_replenish
     UserNotifyService.wish_list_arrival(item_spec) if amount_changed? && amount > 0
   end
+
+  def print_changed_record_to_stock_spec_logger
+    before_changed_amount = amount_was
```
Comment (varbark) [inline app/models/stock_spec.rb]: 覺得原本的amount_was 就夠清楚了

File: app/models/stock_spec.rb [hunk 2]
```diff
@@ -39,4 +40,12 @@ def set_on_shelf_and_send_wish_list_notification_when_amout_larger_than_zero
...
+    increments = amount - before_changed_amount
+    operation = increments > 0 ? "+" : "-"
```
Comment (varbark) [inline app/models/stock_spec.rb]: how about just:    operation = amount > amount_was ? "+" : "-"

File: app/models/stock_spec.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@ class StockSpec < ActiveRecord::Base
 
   after_update :sef_item_spec_off_shelf_if_stock_empty_and_stop_replenish, :send_wish_list_notification_if_stock_replenish
   after_create :set_on_shelf_and_send_wish_list_notification_when_amout_larger_than_zero
+  before_save :print_changed_record_to_stock_spec_logger
```
Comment (StevenKo) [inline app/models/stock_spec.rb]: log_xxxxxxx

File: app/models/stock_spec.rb (hunk 2 above)
Comment (StevenKo) [reply app/models/stock_spec.rb]: Action: #{operation}#{increments.abs} 為什麼不在這裡就處理好？

**Comment by StevenKo** [comment]
要再加 item_id

File: app/models/stock_spec.rb [hunk 4]
```diff
@@ -39,4 +40,11 @@ def set_on_shelf_and_send_wish_list_notification_when_amout_larger_than_zero
   def send_wish_list_notification_if_stock_replenish
     UserNotifyService.wish_list_arrival(item_spec) if amount_changed? && amount > 0
   end
+
+  def log_stock_spec_amount_change
+    increments = amount - amount_was
+    operation = increments > 0 ? "+#{increments}" : "-#{increments.abs}"
```
Comment (StevenKo) [inline app/models/stock_spec.rb]: operation = increments > 0 ? "+#{increments}" : "#{increments}"
??

**Pull request merged**